api/allocations_test.go

```diff
@@ -247,6 +249,58 @@ func TestAllocations_RescheduleInfo(t *testing.T) {
 
 }
 
+// TestAllocations_ExecErrors ensures errors are properly formatted
+func TestAllocations_ExecErrors(t *testing.T) {
+	c, s := makeClient(t, nil, nil)
+	defer s.Stop()
+	a := c.Allocations()
+
+	job := &Job{
+		Name:      stringToPtr("foo"),
+		Namespace: stringToPtr(DefaultNamespace),
+		ID:        stringToPtr("bar"),
+		ParentID:  stringToPtr("lol"),
+		TaskGroups: []*TaskGroup{
+			{
+				Name: stringToPtr("bar"),
+				Tasks: []*Task{
+					{
+						Name: "task1",
+					},
+				},
+			},
+		},
+	}
+	job.Canonicalize()
+
+	uuidGen := func() string {
+		ret, err := uuid.GenerateUUID()
+		if err != nil {
+			t.Fatal(err)
+		}
+		return ret
+	}
```

Comment:
We could switch to using our helper `uuid.Generate()`[1] that panics if fails.  It's used in commonly in test files, but looks like this file and `api/api_test.go` haven't been updated to use it yet.

[1] https://github.com/hashicorp/nomad/blob/v0.10.0/helper/uuid/uuid.go

Reply:
oh nice, will update!